dataengineeringutils3/db.py

```diff
@@ -55,7 +65,11 @@ def __init__(self, cursor, select_query, fetch_size=1000, **query_kwargs):
         self.cursor = cursor
         self.cursor.arraysize = fetch_size
         self.fetch_size = fetch_size
-        self.cursor.execute(select_query, **query_kwargs)
+        if timeout:
```

Comment:
untested, but could something like this work? Then pass `timeout=True` somewhere like [here](https://github.com/moj-analytical-services/airflow-nomis-ap/blob/326e8aca82284b7c2082001252cb2451d2de2872/extraction/nomis_delta_extract.py#L71)

just an idea and probably not the best (might not even work) happy to discuss @gwionap